xbmc/cores/VideoPlayer/VideoPlayerRadioID3.cpp

```diff
@@ -0,0 +1,359 @@
+                 ++ct)
+            {
+              auto commentsFrame = dynamic_cast<const ID3v2::CommentsFrame*>(*ct);
+              if (commentsFrame && commentsFrame->description().isEmpty())
+              {
+                currentMusic->SetComment(commentsFrame->text().to8Bit(true));
+                changed = true;
+                break;
+              }
+            }
+          }
+
+          else if (it->first == "TCON")
+          {
+            currentMusic->SetGenre(GetID3v2StringList(it->second));
+            changed = true;
+          }
+
+          else if (it->first == "TYER")
+          {
+            currentMusic->SetYear(static_cast<int>(
+                strtol(it->second.front()->toString().toCString(true), nullptr, 10)));
+            changed = true;
+          }
+
+          else if (it->first == "TRCK")
+          {
+            currentMusic->SetTrackNumber(static_cast<int>(
+                strtol(it->second.front()->toString().toCString(true), nullptr, 10)));
+            changed = true;
+          }
+
+          // Support for setting the cover art image via CMusicInfoTag does not currently exist,
+          // the code sample below would check for an ID3v2 "APIC" tag of the proper type and
+          // convert the information into an EmbeddedArt object instance
+          //
+          //else if (it->first == "APIC")
+          //{
+          //  // Loop through and look for the FrontCover picture frame
+          //  for (ID3v2::FrameList::ConstIterator pi = it->second.begin(); pi != it->second.end();
+          //       ++pi)
+          //  {
+          //    auto pictureFrame = dynamic_cast<ID3v2::AttachedPictureFrame*>(*pi);
+          //    if (pictureFrame && pictureFrame->type() == ID3v2::AttachedPictureFrame::FrontCover)
+          //    {
+          //      EmbeddedArt coverArt(
+          //          reinterpret_cast<const uint8_t*>(pictureFrame->picture().data()),
+          //          pictureFrame->size(), pictureFrame->mimeType().to8Bit(true));
+          //
+          //      // Assumes "void CMusicInfoTag::SetCoverArt(const EmbeddedArt& art)" exists
+          //      currentMusic->SetCoverArt(coverArt);
+          //      changed = true;
+          //    }
+          //  }
+          //}
+        }
+      }
+
+      if (changed)
+        CServiceBroker::GetGUI()->GetInfoManager().SetCurrentItem(g_application.CurrentFileItem());
+    }
+  }
+}
+
+std::vector<std::string> CDVDRadioID3Data::GetID3v2StringList(const ID3v2::FrameList& frameList)
+{
+  auto frame = dynamic_cast<const ID3v2::TextIdentificationFrame*>(frameList.front());
+  if (frame)
+    return StringListToVectorString(frame->fieldList());
+  return std::vector<std::string>();
+}
+
+std::vector<std::string> CDVDRadioID3Data::StringListToVectorString(const StringList& stringList)
+{
+  std::vector<std::string> values;
+  for (const auto& it : stringList)
```

Comment:
as the var does not hold an iterator, but the de-referenced iterator, I would not name it `it`.